docs/configuration.asciidoc

```diff
@@ -129,6 +129,31 @@ of the deployed revision, e.g. the output of git rev-parse HEAD.
 | Informational version of the entry assembly    | String 
 |============
 
+[float]
+[[config-environment]]
+==== `Environment`
+
+The name of the environment this service is deployed in, e.g. "production" or "staging".
+
+Environments allow you to easily filter data on a global level in the APM UI.
+It's important to be consistent when naming environments across agents.
+See {kibana-ref}/filters.html#environment-selector[environment selector] in the Kibana UI for more information.
+
+NOTE: This feature is fully supported in the APM UI in Kibana versions >= 7.2.
+You must use the query bar to filter for a specific environment in versions prior to 7.2.
+
+[options="header"]
+|============
+| Environment variable name      | IConfiguration key 
+| `ELASTIC_APM_ENVIRONMENT`      | `ElasticApm:Environment`
+|============
+
+[options="header"]
+|============
+| Default                                        | Type
+| https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.ihostingenvironment.environmentname?view=aspnetcore-2.2#Microsoft_AspNetCore_Hosting_IHostingEnvironment_EnvironmentName[EnvironmentName from IHostingEnvironment] | String 
```

Comment:
Thank for the docs :)

It might worth adding that this default is only for ASP.NET Core applications - I don't it will fit well in that table cell so I would put in the table cell a phrase like "See note below" and then in the note below the table clarification that `EnvironmentName` from `IHostingEnvironment` is the default for ASP.NET Core applications.

Reply:
fixed